docs/source/basics.rst

```diff
@@ -273,14 +273,83 @@ be executed 1000 times.
 
 Parametric Compilation
 ~~~~~~~~~~~~~~~~~~~~~~
-*Coming soon*
+
+Keri to improve:
```

Comment:
Modern quantum algorithms are often parametric, following a hybrid model. In this hybrid
model, the program ansatz (template of gates) is fixed, and iteratively updated with new
parameters. These new parameters are often determined by an update given by a classical optimizer
. Depending on the complexity of algorithm, the problem of interesting, and capabilities of the
classical optimizer, this loop may need to run many times. In order to efficiently operate within
 this hybrid model, parametric compilation can be used.

Parametric compilation allows one to compile the program ansatz just once. Making use of declared
 memory regions, we can load values to the parametric gates at execution time, after compilation.
 Taking the compiler out of the execution loop for programs like this offers a huge performance
 improvement compared to compiling the program each time parameters are updated. (Some more
 details about this and an example are found :ref:`here <parametric>`.)